cashier-google-play-billing/src/main/java/com/getkeepsafe/cashier/billing/GooglePlayBillingVendor.java

```diff
@@ -0,0 +1,500 @@
+
+    private boolean available = false;
+    private boolean initializing = false;
+    private boolean canSubscribe = false;
+    private boolean canPurchaseItems = false;
+
+    /**
+     * Tokens to be consumed. Contains token being currently consumed or already consumed.
+     */
+    private Set<String> tokensToBeConsumed = new HashSet<>();
+
+    public GooglePlayBillingVendor() {
+        this(new GooglePlayBillingApi(), null);
+    }
+
+    public GooglePlayBillingVendor(String publicKey64) {
+        this(new GooglePlayBillingApi(), publicKey64);
+    }
+
+    public GooglePlayBillingVendor(AbstractGooglePlayBillingApi api) {
+        this(api, null);
+    }
+
+    public GooglePlayBillingVendor(AbstractGooglePlayBillingApi api, @Nullable String publicKey64) {
+        Preconditions.checkNotNull(api, "Cannot initialize will null api...");
+
+        this.api = api;
+        this.publicKey64 = publicKey64;
+        available = false;
+    }
+
+    @Override
+    public String id() {
+        return VENDOR_PACKAGE;
+    }
+
+    @Override
+    public synchronized void initialize(Context context, InitializationListener listener) {
+        Preconditions.checkNotNull(context, "Cannot initialize with null context");
+        Preconditions.checkNotNull(listener, "Cannot initialize with null initialization listener");
+
+        if (available()) {
+            listener.initialized();
+            return;
+        }
+
+        initializationListeners.add(listener);
+
+        if (!initializing) {
+            initializing = true;
+            logSafely("Initializing Google Play Billing API...");
+            available = api.initialize(context, this, this, logger);
+        }
+
+        if (!available) {
+            initializationListeners.remove(listener);
+            listener.unavailable();
+        }
+    }
+
+    @Override
+    public synchronized void initialized(boolean success) {
+        logSafely("Initialized: success = " + success);
+        if (!success) {
+            logAndDisable("Could not create Google Play Billing instance");
+            return;
+        }
+
+        try {
+            canPurchaseItems =
+                    api.isBillingSupported(SkuType.INAPP) == BillingResponse.OK;
+
+            canSubscribe =
+                    api.isBillingSupported(SkuType.SUBS) == BillingResponse.OK;
+
+            available = canPurchaseItems || canSubscribe;
+            logSafely("Connected to service and it is " + (available ? "available" : "not available"));
+            initializing = false;
+
+            for (InitializationListener listener : initializationListeners) {
+                listener.initialized();
+            }
+        } catch (Exception error) {
+            logAndDisable(Log.getStackTraceString(error));
+        }
+    }
+
+    @Override
+    public void disconnected() {
+        logAndDisable("Disconnected from Google Play Billing service.");
```

Comment:
Should the `initializationListeners` `#unavailable` should be invoked here and also unregistered as not to leak its reference?